src/main/java/org/stroganov/ws/handlers/RequestsValidator.java

```diff
@@ -0,0 +1,124 @@
+package org.stroganov.ws.handlers;
+
+import jakarta.xml.soap.*;
+import jakarta.xml.ws.handler.MessageContext;
+import jakarta.xml.ws.handler.soap.SOAPHandler;
+import jakarta.xml.ws.handler.soap.SOAPMessageContext;
+import jakarta.xml.ws.soap.SOAPFaultException;
+import org.apache.log4j.Logger;
+import org.stroganov.dao.LibraryDAO;
+import org.stroganov.entities.User;
+import org.stroganov.util.DataManager;
+
+import javax.xml.namespace.QName;
+import java.util.*;
+
+public class RequestsValidator implements SOAPHandler<SOAPMessageContext> {
+    private final Logger logger = Logger.getLogger(RequestsValidator.class);
+    private final List<String> adminRequests = Arrays.asList("ns2:deleteUser", "ns2:blockUser", "ns2:unblockUser", "ns2:getAllHistory", "ns2:addUser");
```

Comment:
such a hardcore code is absolutely not acceptable ( rename or adding new function will break your code)
```suggestion
 remove
```

Reply:
I've replaced this list in new Enum. What is another way to solve such situations ?

Reply:
Just create a new handler, and mark class responsible for admin action with it